// unplugin-vue-define-options/dist/chunk-XNTJX2KA.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _chunkXUCCRJPTjs = require('./chunk-XUCCRJPT.js');

// src/index.ts
var _unplugin = require('unplugin');
var _pluginutils = require('@rollup/pluginutils');
function resolveOption(options) {
  return {
    include: options.include || [/\.vue$/, /\.vue\?vue/],
    exclude: options.exclude || void 0
  };
}
var src_default = _unplugin.createUnplugin.call(void 0, (options = {}) => {
  const opt = resolveOption(options);
  const filter = _pluginutils.createFilter.call(void 0, opt.include, opt.exclude);
  const name = "unplugin-vue-define-options";
  return {
    name,
    enforce: "pre",
    transformInclude(id) {
      return filter(id);
    },
    transform(code, id) {
      try {
        const s = _chunkXUCCRJPTjs.transform.call(void 0, code, id);
        if (!s)
          return;
        return {
          code: s.toString(),
          get map() {
            return s.generateMap();
          }
        };
      } catch (err) {
        this.error(`${name} ${err}`);
      }
    }
  };
});



exports.src_default = src_default;
